lib/nrf_modem_lib/nrf91_net_l2/ip_addr_helper.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @brief Add IPv4 address for the default PDP context to the network interface.
+ *
+ *  @param Pointer to network interface.
+ *
+ *  @retval 0 on success.
+ *  @retval -EINVAL if the passed in network interface pointer was NULL.
+ *  @retval -ENODATA if no IPv4 address was obtained from the modem.
+ *  @retval -EFAULT if an error occurred when obtaining the IPv4 address.
+ *  @retval -ENODEV if the passed in interface pointer was not valid.
+ */
+int ipv4_addr_add(const struct net_if *iface);
+
+/** @brief Add IPv6 address for the default PDP context to the network interface.
+ *
+ *  @param Pointer to network interface.
+ *
+ *  @retval 0 on success.
+ *  @retval -EINVAL if the passed in network interface pointer was NULL.
+ *  @retval -ENODATA if no IPv6 address was obtained from the modem.
+ *  @retval -EFAULT if an error occurred when obtaining the IPv6 address.
+ *  @retval -ENODEV if the passed in interface pointer was not valid.
+ */
+int ipv6_addr_add(const struct net_if *iface);
+
+/** @brief Remove IPv4 address that was previously associated with the default PDP context
+ *         from the network interface.
+ *
+ *  @param Pointer to network interface.
+ *
+ *  @retval 0 on success or there was no IPv4 address to remove.
+ *  @retval -EINVAL if the passed in network interface pointer was NULL.
+ *  @retval -EFAULT if the IPv4 address failed to be removed.
+ */
+int ipv4_addr_remove(const struct net_if *iface);
+
+/** @brief Remove IPv6 address that was previously associated with the default PDP context
+ *         from the network interface.
+ *
+ *  @param Pointer to network interface.
+ *
+ *  @retval 0 on success or there was no IPv6 address to remove.
+ *  @retval -EINVAL if the passed in network interface pointer was NULL.
+ *  @retval -EFAULT if the IPv6 address failed to be removed.
+ */
+int ipv6_addr_remove(const struct net_if *iface);
```

Comment:
Well, these files and functions are local and contained in a folder named `nrf91_net_l2` so by extention they are related. I think it apparent enough when factoring in the context. I don't usually see the need of prexing helper functions and static functions that are used in only one place.

Reply:
Mainly my concern is that they look very similar to existing Zephyr definitions. For instance `ipv6_addr_add` vs `net_if_ipv6_addr_add`, so it harms readability. Especially since `ipv6_addr_add` is not defined in the same `.c` file it is used in. Plus prefixing with `nrf91` is a bit truer to what they actually do, since they are helpers that are specifically specialized for interacting with `nrf91` AT commands

But if you strongly prefer to keep them named as they are, I won't block the PR over it, it's ultimately your decision since you are the maintainer, or at least the author, of this code - it ought to be your final say on what you name things